xonsh/aliases.py

```diff
@@ -254,7 +254,8 @@ def source_cmd(args, stdin=None):
     args[0] = fpath if fpath else args[0]
     if not os.path.isfile(args[0]):
         raise FileNotFoundError(args[0])
-    prevcmd = 'call "{}"\necho off'.format(args[0])
+    prevcmd = 'call "{}"'.format('" "'.join(args))
```

Comment:
Interesting but you are probably right. Btw. I believe `Popen` only takes care of the ARGV part, not the CMD part but I may be wrong.